testing/test_segment_volume.py

```diff
@@ -0,0 +1,105 @@
+import os
+import json
+import torch
+import nibabel as nib
+import shutil
+
+from ivadomed import utils as imed_utils
+from ivadomed import models as imed_models
+
+SLICE_AXIS = 2
+PATH_BIDS = 'testing_data'
+PATH_MODEL = os.path.join(PATH_BIDS, "model_test")
+IMAGE_PATH = os.path.join(PATH_BIDS, "sub-test001", "anat", "sub-test001_T1w.nii.gz")
+ROI_PATH = os.path.join(PATH_BIDS, "derivatives", "labels", "sub-test001", "anat",
+                        "sub-test001_T1w_seg-manual.nii.gz")
+BATCH_SIZE = 1
+DROPOUT = 0.4
+BN = 0.1
+LENGTH_3D = [96, 96, 16]
+
+
+def test_segment_volume_2d():
+    model = imed_models.Unet(in_channel=1,
+                             out_channel=1,
+                             depth=2,
+                             drop_rate=DROPOUT,
+                             bn_momentum=BN)
+
+    # temporary folder that will be deleted at the end of the test
+    if not os.path.exists(PATH_MODEL):
+        os.mkdir(PATH_MODEL)
+
+    torch.save(model, os.path.join(PATH_MODEL, "model_test.pt"))
+    config = {
+        "batch_size": BATCH_SIZE,
+
+        "transformation_validation": {
+            "Resample": {"wspace": 0.75, "hspace": 0.75},
+            "ROICrop2D": {"size": [48, 48]},
+            "ToTensor": {},
+            "NormalizeInstance": {}
+        },
+        "slice_filter": {
+            "filter_empty_mask": False,
+            "filter_empty_input": False
+        },
+        "slice_filter_roi": 10,
+        "slice_axis": "axial",
+        "unet_3D": False,
+    }
+    PATH_CONFIG = os.path.join(PATH_MODEL, 'model_test.json')
+    with open(PATH_CONFIG, 'w') as fp:
+        json.dump(config, fp)
+
+    nib_img = imed_utils.segment_volume(PATH_MODEL, IMAGE_PATH, ROI_PATH)
+    assert nib_img.shape == nib.load(IMAGE_PATH).shape
```

Comment:
Add data type test? plus range of values (not sure if it will work with init_Unet)?

Reply:
Good idea! I'll add it :)

Reply:
For the data type, segment_volume returned float64, which is not necessary. I forced the output to be float32 in pred_to_nib since it's the datatype used in the whole pipeline (never float64). Also, float32 rather than intXX because we still want the option to have a soft output (our value range is between 0 and 1, so casting in int returns a hard mask).